src/installer/tests/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/ResourceUpdaterTests.cs

```diff
@@ -0,0 +1,37 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.IO;
+using System.Reflection.PortableExecutable;
+using ILCompiler.Win32Resources;
+using Microsoft.DotNet.CoreSetup.Test;
+using Microsoft.NET.HostModel;
+using Xunit;
+
+namespace AppHost.Bundle.Tests;
+
+public class ResourceUpdaterTests
+{
+    RepoDirectoriesProvider RepoDirectories = new RepoDirectoriesProvider();
+
+    [WindowsOnlyFact]
+    void ResourceUpdaterAddResource()
+    {
+        using var memoryStream = new MemoryStream(File.ReadAllBytes(Path.Combine(RepoDirectories.HostArtifacts, "comhost.dll")));
```

Comment:
Test for `ResourceUpdater` needs some valid PE file with `.rsrc` section so I choose comhost.dll without any reason for now. 
However, I think this is not good because base PE file should be build for platforms other than windows and test should be ran on those platforms but I could not find what's the best one.

Reply:
apphost (and singlefilehost) would be the common product scenario - `Binaries.AppHost.FilePath` should have the path to the live-built executable. Those actually intentionally don't include any resources (they represent the user's app, so should only have any resources specified by the user), so wouldn't have a `.rsrc` section.

Testing using apphost on non-Windows would be a bit more challenging, since our CI/PR builds running on non-Windows won't have built the Windows apphost. For unit tests, you could:
- Use the current assembly (since .NET binaries are conformant PE images) - this would have a resource section
- Use [`ManagedPEBuilder`](https://learn.microsoft.com/dotnet/api/system.reflection.portableexecutable.managedpebuilder) to create a minimal PE image with no resource section. Something like:
  ```c#
  var peBuilder = new ManagedPEBuilder(
      PEHeaderBuilder.CreateExecutableHeader(),
      new MetadataRootBuilder(new MetadataBuilder()),
      ilStream: new BlobBuilder());
  var peImageBuilder = new BlobBuilder();
  peBuilder.Serialize(peImageBuilder);
  byte[] peBytes = peImageBuilder.ToArray();
  ```